mirrord-agent/src/file.rs

```diff
@@ -288,6 +297,32 @@ impl FileManager {
             })
     }
 
+    #[tracing::instrument(level = "trace", skip(self))]
+    pub(crate) fn write_limited(
+        &mut self,
+        fd: usize,
+        start_from: u64,
+        buffer: Vec<u8>,
+    ) -> RemoteResult<WriteFileResponse> {
+        self.open_files
+            .get_mut(&fd)
+            .ok_or(ResponseError::NotFound(fd))
+            .and_then(|remote_file| {
+                if let RemoteFile::File(file) = remote_file {
+                    let mut writer = BufWriter::new(std::io::Write::by_ref(file));
+                    let _ = writer.seek(SeekFrom::Start(start_from))?;
+
+                    let write_result = writer
+                        .write(&buffer)
+                        .map(|written_amount| WriteFileResponse { written_amount })?;
+
```

Comment:
Remember to call [`flush`](https://doc.rust-lang.org/std/io/struct.BufWriter.html#method.flush) as the documentation [says it's critical](https://doc.rust-lang.org/std/io/struct.BufWriter.html#method.flush) to do so before dropping.